selftests/run

```diff
@@ -26,6 +35,11 @@ def test_suite():
     for section in ('unit', 'functional', 'doc'):
         suite.addTests(loader.discover(start_dir=os.path.join(selftests_dir, section),
                                        top_level_dir=basedir))
+    # when other plugins get their own tests, make this a generic loop
+    if plugin_available('avocado-framework-plugin-varianter-yaml-to-mux'):
+        path = os.path.join(basedir, 'optional_plugins', 'varianter_yaml_to_mux')
+        suite.addTests(loader.discover(start_dir='tests',
+                                       top_level_dir=path))
```

Comment:
This does not cover the `PARALLEL` or `AVOCADO` check.

Reply:
The more I think about this I'd prefer to put optional_plugins tests into `selftests/optional_plugins/` directory. It won't be perfect from separation point of view, but it'd be way simpler to maintain and use, besides I don't think we could expect many optional-plugin selftests and if so the optional-plugin would probably be so complex that at that point it'd already be a separate entity like `avocado-virt`. What do you think?

Reply:
And the more I look at our custom shell/Python scripts to run our own tests, the more I think we are rewriting (non-optimally) what Avocado is meant to be doing.  Running tests in parallel is one thing.  Finding tests across directories is another one.

Anyway, I do think we should attempt to make plugins really that: separated and optional, including tests.  I'll address the other use cases such as PARALLEL.

Reply:
yep, when Avocado has support to run things in parallel, we could simplify our execution, but until then parallel is important to me... (It's really fast with 8 cores)

Reply:
Then please take a look at this:

https://github.com/avocado-framework/avocado/pull/2358